1.57 new file lib/nghttp2_time.c builds fail on Cygwin and Mingw

New file in *1.57* `lib/nghttp2_time.c` builds fail on *Cygwin* and *Mingw* cross as both `GetTickCount64`() and `clock_gettime`() are defined, and recent fix #1963 *Fix build error when both clock_gettime and GetTickCount64 are available* for #1962 *compilation error on *Windows* with *MSYS2* : undefined reference to `time_now_sec`* does not handle the issue comprehensively; we want to use the *POSIX* interface for *Cygwin* builds, but the *Windows* interface for *Mingw* (cross-)builds. 
The patch below resolves the issue for *Cygwin* and *Mingw* cross, and hopefully also other *POSIX* and *Windows*/*Msys2*, but I don't have build environments to test those: additional `!`/`defined()`s *may* be needed after review:
[nghttp2-1.57-nghttp2_time-c-time_now_sec-cygwin-mingw-defines.patch](https://github.com/nghttp2/nghttp2/files/12931873/nghttp2-1.57-nghttp2_time-c-time_now_sec-cygwin-mingw-defines.patch)

Please also consider including the effects of the additional patches below in some way after review: to `configure` *C++* `-std=gnu...` for use with stricter feature test macro settings; and to use static lib in `Makefile` for building tests; these have been used since before I adopted maintaining *nghttp2* for *Cygwin* & *Mingw* cross under *Cygwin*:
[configure-ac-cxx-ext.patch](https://github.com/nghttp2/nghttp2/files/12932784/configure-ac-cxx-ext.patch)
[tests-Makefile-am-staticlib.patch](https://github.com/nghttp2/nghttp2/files/12932853/tests-Makefile-am-staticlib.patch)

--- a/lib/nghttp2_time.c
+++ b/lib/nghttp2_time.c
@@ -32,7 +32,7 @@
 #  include <sysinfoapi.h>
 #endif /* HAVE_SYSINFOAPI_H */
 
-#ifndef HAVE_GETTICKCOUNT64
+#if !defined(HAVE_GETTICKCOUNT64) || defined(__CYGWIN__)
 static uint64_t time_now_sec(void) {
   time_t t = time(NULL);
 
@@ -42,9 +42,9 @@ static uint64_t time_now_sec(void) {
 
   return (uint64_t)t;
 }
-#endif /* HAVE_GETTICKCOUNT64 */
+#endif /* !HAVE_GETTICKCOUNT64 || __CYGWIN__ */
 
-#ifdef HAVE_GETTICKCOUNT64
+#if defined(HAVE_GETTICKCOUNT64) && !defined(__CYGWIN__)
 uint64_t nghttp2_time_now_sec(void) { return GetTickCount64() / 1000; }
 #elif defined(HAVE_CLOCK_GETTIME)
 uint64_t nghttp2_time_now_sec(void) {
@@ -57,6 +57,6 @@ uint64_t nghttp2_time_now_sec(void) {
 
   return (uint64_t)tp.tv_sec;
 }
-#else  /* !HAVE_CLOCK_GETTIME && !HAVE_GETTICKCOUNT64 */
+#else  /* (!HAVE_CLOCK_GETTIME || __CYGWIN__) && !HAVE_GETTICKCOUNT64 */
 uint64_t nghttp2_time_now_sec(void) { return time_now_sec(); }
-#endif /* !HAVE_CLOCK_GETTIME && !HAVE_GETTICKCOUNT64 */
+#endif /* (!HAVE_CLOCK_GETTIME || __CYGWIN__) && !HAVE_GETTICKCOUNT64 */

